forge/db.py

```diff
@@ -28,20 +28,74 @@
 # Create pickable object
 class PopulateFeaturesArguments:
 
-    def __init__(self, engineURL, modelIndex, shpFile):
+    def __init__(self, engineURL, modelIndex, shpFile, reproject):
         self.engineURL = engineURL
         self.modelIndex = modelIndex
         self.shpFile = shpFile
+        self.reproject = reproject
+
+
+def reprojectShp(shpFilePath):
+    logger.info('Action reprojectShapefile(%s)' % shpFilePath)
+    outDirectory = config.get('Reprojection', 'outDirectory')
+    outFile = '%s%s' % (outDirectory, os.path.basename(shpFilePath))
+
+    # If out file already exists clean it up first
+    cleanup(outFile)
+
+    command = '%(geosuiteCmd)s -calc reframe -in %(inFile)s -out %(outFile)s -pframes %(fromPFrames)s,%(toPFrames)s ' \
+        '-aframes %(fromAFrames)s,%(toAFrames)s -log %(logfile)s -err %(errorfile)s' % dict(
+            geosuiteCmd=config.get('Reprojection', 'geosuiteCmd'),
+            inFile=shpFilePath,
+            outFile=outFile,
+            fromPFrames=config.get('Reprojection', 'fromPFrames'),
+            toPFrames=config.get('Reprojection', 'toPFrames'),
+            fromAFrames=config.get('Reprojection', 'fromAFrames'),
+            toAFrames=config.get('Reprojection', 'toAFrames'),
+            logfile=config.get('Reprojection', 'logfile'),
+            errorfile=config.get('Reprojection', 'errorfile')
+        )
+    try:
+        logger.info('Command: %s' % command)
+        subprocess.call(command, shell=True)
+    except Exception as e:
+        logger.error('Could not reproject %(inFile)s into %(outFile)s: %(err)s' % dict(
+            inFile=shpFilePath,
+            outFile=outFile,
+            err=e
+        ))
+        raise Exception(e)
+
+    # As we can't detect a success from an error with the current implementation
```

Comment:
newer version of mono (version 3.2.8 (Debian 3.2.8+dfsg-10)) returns a status code different from zero if there is some trouble.
```bash
ltclm@ip-10-220-6-191:~$ mono /home/geodata/GeoSuiteCmdx64/GeoSuiteCmd.exe
swisstopo GeoSuite console x64 1.3
Calculation module name missing or invalid
ltclm@ip-10-220-6-191:~$ echo $?
160
```
@loicgasser please test the reprojection script on our db instance, i have installed mono there.
The geodesy tool is already installed under ``/home/geodata/GeoSuiteCmdx64/GeoSuiteCmd.exe``